op_crates/fetch/22_body.js

```diff
@@ -0,0 +1,338 @@
+      if (object[bodySymbol] !== null) {
+        return object[bodySymbol].consume();
+      }
+      return Promise.resolve(new Uint8Array());
+    }
+
+    /** @type {PropertyDescriptorMap} */
+    const mixin = {
+      body: {
+        /**
+         * @returns {ReadableStream<Uint8Array> | null}
+         */
+        get() {
+          webidl.assertBranded(this, prototype);
+          if (this[bodySymbol] === null) {
+            return null;
+          } else {
+            return this[bodySymbol].stream;
+          }
+        },
+      },
+      bodyUsed: {
+        /**
+         * @returns {boolean}
+         */
+        get() {
+          webidl.assertBranded(this, prototype);
+          if (this[bodySymbol] !== null) {
+            return this[bodySymbol].consumed();
+          }
+          return false;
+        },
+      },
+      arrayBuffer: {
+        /** @returns {Promise<ArrayBuffer>} */
+        value: async function arrayBuffer() {
+          webidl.assertBranded(this, prototype);
+          const body = await consumeBody(this);
+          return packageData(body, "ArrayBuffer");
+        },
+      },
+      blob: {
+        /** @returns {Promise<Blob>} */
+        value: async function blob() {
+          webidl.assertBranded(this, prototype);
+          const body = await consumeBody(this);
+          return packageData(body, "Blob", this[mimeTypeSymbol]);
+        },
+      },
+      formData: {
+        /** @returns {Promise<FormData>} */
+        value: async function formData() {
+          webidl.assertBranded(this, prototype);
+          const body = await consumeBody(this);
+          return packageData(body, "FormData", this[mimeTypeSymbol]);
+        },
+      },
+      json: {
+        /** @returns {Promise<any>} */
+        value: async function json() {
+          webidl.assertBranded(this, prototype);
+          const body = await consumeBody(this);
+          return packageData(body, "JSON");
+        },
+      },
+      text: {
+        /** @returns {Promise<string>} */
+        value: async function text() {
+          webidl.assertBranded(this, prototype);
+          const body = await consumeBody(this);
+          return packageData(body, "text");
+        },
+      },
+    };
+    return Object.defineProperties(prototype.prototype, mixin);
+  }
+
+  const decoder = new TextDecoder();
+
+  /**
+   * https://fetch.spec.whatwg.org/#concept-body-package-data
+   * @param {Uint8Array} bytes
+   * @param {"ArrayBuffer" | "Blob" | "FormData" | "JSON" | "text"} type
+   * @param {MimeType | null} [mimeType]
+   */
+  function packageData(bytes, type, mimeType) {
+    switch (type) {
+      case "ArrayBuffer":
+        return bytes.buffer;
+      case "Blob":
+        return new Blob([bytes], {
+          type: mimeType !== null ? mimesniff.serializeMimeType(mimeType) : "",
+        });
+      case "FormData": {
+        if (mimeType !== null) {
+          if (mimeType !== null) {
```

Comment:
Really late, but the nested `if` statement body is unreachable, you have a duplicate condition.